### DISCUSSION TITLE
a chrome extension as a more accessible interface for shortest

### DISCUSSION THREAD

**Discussion by amk-dev** [antiwork/shortest#175] [Ideas]
* chrome sidebar
* create / manage tests from the sidebar, no writing code required ?
* can deploy the selected tests to shortest to run in ci
* no dependency on any codebase, come to a webpage, write tests for it, run it whenever you want

<img width="1423" alt="image" src="https://github.com/user-attachments/assets/2ac3772b-23c4-4f8d-8f9b-9d4e4323af1c" />

<img width="1256" alt="image" src="https://github.com/user-attachments/assets/7839d7e5-c707-4415-b22d-fc3e13f270e5" />

these are not designs, i just put some text in a screenshot to make the point.

here's a realworld screenshot from a chrome extension i'm working on

<img width="1001" alt="image" src="https://github.com/user-attachments/assets/acebc469-a7d2-408a-b1f8-a6f9ccdbd80c" />

**Comment by slavingia** [comment]
Interesting idea though I'd rather it be part of the codebase and run on every pr automatically via CI

**Reply by amk-dev** [reply]
got it.

thinking out loud on this idea,

1. user downloads the extension
2. goes to any website that they'd like (eg: `staging.their-product.com`)
    1. write tests for the website
    2. they can come to the extension and rerun the tests whenever they want, ( the same tests can be run in multiple websites, (eg: `testing.their-product.com` / `their-product.com`)) )
    2. or they can push them to `shortest cloud`
        1. when you save the tests to shortest cloud, you'll get a `testsuiteID` / `slug`
        2. which you can use to run the tests from your CI 
            1. something like `shortest run your-testsuite-slug`, which would trigger the tests in the cloud 
            2. or if you want to run them in the github actions for eg: pull the tests from shortest cloud and run them in your environment, `shortest run your-testsuite-slug --local`

this way may be shortest cloud can be monetized.
- small startups might like this, because a lot of them do manual testing when releasing.
- no bandwidth from engineering needed ( apart from setting up a ci workflow, may be we can write a `shortest github action` that makes this easy ) 

also the scope is kinda big, and some unknowns are there. but i can give it a go if you're seeing some potential in this

**Reply by slavingia** [reply]
IMO not worth working on until the main code package has traction!

**Reply by amk-dev** [reply]
yeah yeah. makes sense
